### ISSUE TITLE
Suggestions on code quality

### ISSUE THREAD

**Issue by stevemonaco** [NerdyMusician/GAMMA#4]
Hi, congrats on making it to the open source step. Some suggestions on the most egregious culprits.

1. The INPC properties are way too long because your `BaseModel` implementation is not good enough to remove more boilerplate. The first option is to add a `SetAndNotify` method to the `BaseClass` like so:

```cs
protected bool SetAndNotify<T>(ref T field, T value, [CallerMemberName] string propertyName = null)
{
	if (EqualityComparer<T>.Default.Equals(field, value))
		return false;

	field = value;
	NotifyPropertyChanged(propertyName);
	return true;
}
```

It prevents infinite cyclic updates and your properties can look like the following after ditching the regions (which I also suggest):

```cs
private string _isBlinded_Tooltip;
public string IsBlinded_Tooltip
{
	get => _isBlinded_Tooltip;
	set => SetAndNotify(ref _isBlinded_Tooltip, value);
}
```

Alternatively, you can use an Mvvm Framework like `CommunityToolkit.Mvvm` which has source generators and can reduce many of your properties down to:

```cs
[ObservableProperty] string _isBlinded_Tooltip;
```

Or at worst, a similar 6-liner as above if you choose to not use source generators.

2. `[XmlSaveMode("Single")]` really should take an enum instead of a string to prevent errors. Alternatively, you may want to create specific types for serialization or project to anonymous types before serializing so that your VMs don't need these attributes at all, but that's a preference.

In general, there's a lot of logic based on strings instead of a more encoded type. This will eventually lead to maintenance issues and/or bugs.

3. The project is on .NET Framework 4.7.1. I assume there's a reason for this (if not, move to .NET 6). At the minimum, you should upgrade the older csproj to the newer SDK-style csproj.

**Comment by NerdyMusician** [comment]
Thank you for the feedback, this will definitely give me things to work on this weekend. 

There isn't a specific reason it is on .NET framework 4.7.1, it's just what I was pointed to when I started learning C# a few years back and didn't have a reason to change, but will look into the process of making that move.

I'll definitely look into the CommunityToolkit.Mvvm framework, as it would be nice to remove excess lines where they aren't needed.

**Comment by NerdyMusician** [comment]
Target .NET version has been updated to .NET 6 and conflicts arising during my testing have been addressed. I opted for the 6 line method utilizing the SetAndNotify<T> option you presented, as various other changes would need to be made in how my application is structured for the CommunityToolkit.Mvvm option to have worked out. There's still quite a few object properties pending the changeover, but as is I was able to remove a few thousand lines of code.
